### DISCUSSION TITLE
[경훈] assignment 기능구현

### DISCUSSION THREAD

**Discussion by tirhande** [Wanted-07-team-9/pre-onboarding-7th-2-1-9#3] [기능구현]
![badge](https://img.shields.io/badge/React-61dafb?logo=React&logoColor=white&style=flat-square) ![badge](https://img.shields.io/badge/Next.js-000000?style=flat-square&logo=Next.js&logoColor=white) ![badge](https://img.shields.io/badge/npm-CB3837?logo=npm&logoColor=white&style=flat-square)

> pre-onboarding-7th-2-1-9

## **Description**

차량 대여를 위해 차량 목록을 불러와서 표시해주는 서비스입니다.  
Next.js를 사용하여 제작하였습니다.

## **Demo Page**  
# [데모 페이지 바로가기](https://pre-onboarding-7th-2-1-9.vercel.app)

## **Assignment 1**

- Figma 상의 디자인 및 기능 구현
  - 모바일 웹 기준으로 제작
  - **450px ~ 360px**까지 고려해서 제작
- 필수 요구 사항
  - Figma 상의 디자인 및 기능 구현

|                                                   Google Chrome                                                    |                                                   iPhone Safari                                                    |
| :-------------------------------------------------------------------------------------------------------------------------: | :-------------------------------------------------------------------------------------------------------------------------: |
| <img src="https://user-images.githubusercontent.com/74575497/199493615-50c635bf-175b-4bad-9698-7a97cd7b451d.gif" width=300> | <img src="https://user-images.githubusercontent.com/74575497/199493644-a9460d6b-9c18-444d-af08-bcd2b7667159.gif" width=200> |

## **Assignment 2**

- 차량 리스트

  |                                                                차량이 없을 때 처리                                                                |                                                               차량 불러오는 중 처리                                                               |
  | :-----------------------------------------------------------------------------------------------------------------------------------------------: | :-----------------------------------------------------------------------------------------------------------------------------------------------: |
  | <p align="center"><img src="https://user-images.githubusercontent.com/74575497/199498073-2eb2b5c9-7c87-4587-8a75-a1934135e9de.png" width=200></p> | <p align="center"><img src="https://user-images.githubusercontent.com/74575497/199497026-f4102819-acba-4fd8-b179-9b51750a17c9.gif" width=200></p> |

- 차량이 없을 때 처리
https://github.com/Wanted-07-team-9/pre-onboarding-7th-2-1-9/blob/efe77d1b4737c8a1c2eabcb54631a652fee4d1b4/src/container/Main/component/CarSection.jsx#L7-L21

- 차량 불러오는 중 처리
https://github.com/Wanted-07-team-9/pre-onboarding-7th-2-1-9/blob/efe77d1b4737c8a1c2eabcb54631a652fee4d1b4/src/container/Main/Main.jsx#L7-L15

## **Assignment 3**

- 차랑 상세

<p align="center">
<img src="https://user-images.githubusercontent.com/74575497/199497550-ca9526b3-9b09-4102-af71-f275c743153b.png" width=200>
</p>

https://github.com/Wanted-07-team-9/pre-onboarding-7th-2-1-9/blob/efe77d1b4737c8a1c2eabcb54631a652fee4d1b4/src/container/Detail/Detail.jsx#L15-L22

[Summary 컴포넌트 링크](https://github.com/Wanted-07-team-9/pre-onboarding-7th-2-1-9/blob/efe77d1b4737c8a1c2eabcb54631a652fee4d1b4/src/container/Detail/component/Summary.jsx)

[SubDescription 컴포넌트 링크](https://github.com/Wanted-07-team-9/pre-onboarding-7th-2-1-9/blob/efe77d1b4737c8a1c2eabcb54631a652fee4d1b4/src/container/Detail/component/SubDescription.jsx)

## **Assignment 4 (추가 구현 사항)**

- **SEO 구현**
- 카카오톡, 페이스북에 공유 시 아래의 내용이 미리보기로 노출되도록 해야 함
  - 제목: car.brand + car.name
  - 내용: 월 car.amount 원
  - 사진: 차량 사진

|                                                                     Discord                                                                     |                                                                     Kakao                                                                     |Facebook|
| :---------------------------------------------------------------------------------------------------------------------------------------------: | :-------------------------------------------------------------------------------------------------------------------------------------------: |:--:|
| <img width="250" alt="seo_discord" src="https://user-images.githubusercontent.com/74575497/199489718-9a3c3bad-0183-448e-8ac2-f295021e5146.png"> | <img width="271" alt="seo_kakao" src="https://user-images.githubusercontent.com/74575497/199489727-04f6dd93-312d-4ee6-bb69-1f42ecb21fd7.png"> |<img width="250" alt="seo_face" src="https://user-images.githubusercontent.com/74575497/199501362-dd45f033-8a46-46af-9745-001b061cc9d8.png"/>|

https://github.com/Wanted-07-team-9/pre-onboarding-7th-2-1-9/blob/efe77d1b4737c8a1c2eabcb54631a652fee4d1b4/pages/car/%5Bid%5D.jsx#L21-L33

## **Assignment 5 (추가 구현 사항)**
 - 😅 페이지 전환시 부드러운 애니메이션 효과 추가
 - 🥳 Figma에 적용된 Inter 폰트로 사용

**Comment by jun-05** [comment]
- 하루만에 거의 대부분 끝내신게 너무 부러웠습니다.
- nextJS도 하루만에 적용하신게 신기했어요.
- 애니메이션 처리도 그렇지만, 상세페이지 화면에서 id값을 바꿔서 이동한후, 뒤로가기 버튼시 자동적으로 전체화면 리스트로 적용하신것등 세심하다고 느꼈습니다.
- 아쉬운점은 모바일에선 잘되는데, PC에선 스크롤바가 너무 작동이 안되요 ㅋㅋ
- 그리고 같은 태그를 두번 누르는 경우엔 굳이 재로딩을 할 필요가 있나 싶었습니다.

**Comment by crystal993** [comment]
- Next.js를 적용하여 SEO 부분 구현하신 점이 굉장히 대단하다고 생각합니다! 
- 로딩 애니메이션도 굉장히 잘 적용하신 것 같습니다! 
- 아쉬운 점은 경훈님이 Discussion에 올려주신대로 shift + 마우스 스크롤을 해야 스와이프가 작동되는 점이 아쉬웠습니다

**Comment by anotheranotherhoon** [comment]
- Next.js 서버사이드렌더링이 SEO의 장점을 가진다는 이야기는 들었으나 직접 CSR과 Next.js를 비교해보니 어떤 점에서 SEO의 장점을 갖는지 피부로 알 수 있는 좋은 예시가 되었습니다.
-  애니메이션으로 자동차가 출발하는 것, 로딩시 fram 효과 까지 디테일이 느껴집니다!

**Comment by songseul** [comment]
- 추가로 애니메이션 적용 하신 점도 어플리케이션의 퀄리티가 높아보입니당
- next.js 처음 사용 하시는데 거기에 seo까지 구현 하신 것도 좋은 방향 인것 같아요

**Comment by NR0617** [comment]
SEO를 너무 잘 구현하셔서 부러웠어요! 
컴포넌트를 적절하게 나눠서 쓰셔서 가독성이 좋아요
